presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -844,18 +845,23 @@ protected RelationType visitTable(Table table, AnalysisContext context)
             // is this a reference to a WITH query?
             String name = table.getName().getSuffix();
 
-            Query query = context.getNamedQuery(name);
-            if (query != null) {
+            WithQuery withQuery = context.getNamedQuery(name);
+            if (withQuery != null) {
+                Query query = withQuery.getQuery();
                 analysis.registerNamedQuery(table, query);
 
                 // re-alias the fields with the name assigned to the query in the WITH declaration
                 RelationType queryDescriptor = analysis.getOutputDescriptor(query);
```

Comment:
I am not sure about the SQL specifications but if only all or none of selected columns can be aliased, then I can insert a length check here.